core/src/mender-client.c

```diff
@@ -943,7 +943,9 @@ mender_client_update_work_function(void) {
                     mender_log_info("Downloading artifact with id '%s', name '%s', uri '%s'", deployment->id, deployment->artifact_name, deployment->uri);
                 }
 #endif
-                mender_client_publish_deployment_status(deployment->id, MENDER_DEPLOYMENT_STATUS_DOWNLOADING);
+                if (MENDER_ABORTED == mender_client_publish_deployment_status(deployment->id, MENDER_DEPLOYMENT_STATUS_DOWNLOADING)) {
+                    ret = MENDER_FAIL;
+                }
```

Comment:
This does nothing, basically, because there's a `ret = mender_download_artifact()` right below which overwrites this value and there's nothing in between to make it skipped.

Reply:
Ah :sweat_smile:, misread it as `MENDER_OK != ret` for some reason, thanks